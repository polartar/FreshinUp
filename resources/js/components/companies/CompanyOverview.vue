<template>
  <v-card>
    <v-card-title class="justify-space-between px-4 py-1">
      <span class="grey--text subheading font-weight-bold">Company overview</span>
      <v-btn
        depressed
        color="primary"
        @click="viewDetails"
      >
        View Details
      </v-btn>
    </v-card-title>
    <v-divider />
    <v-card-text>
      <v-progress-linear
        v-if="isLoading"
        indeterminate
      />
      <v-layout
        v-if="emptyCompany"
        class="pa-4"
        align-center
        justify-center
      >
        Empty company
      </v-layout>
      <v-layout
        v-else
        class="pa-4"
        align-center
        justify-center
      >
        <v-flex class="mr-4">
          <v-img
            :src="companyLogo"
            class="grey lighten-2"
          />
        </v-flex>
        <v-flex>
          <div class="primary--text subheading">
            {{ name }}
          </div>
          <div class="grey--text caption">
            Type: {{ typeName }}
          </div>
          <div class="grey--text caption">
            Status {{ statusName }}
          </div>
        </v-flex>
        <v-flex class="grey--text caption">
          Member {{ members_count }}
        </v-flex>
        <v-flex class="grey--text caption">
          {{ admin.level_name }}
        </v-flex>
        <v-flex>
          <v-layout align-center>
            <v-flex
              xs3
              mx-2
            >
              <f-user-avatar
                :tile="false"
                :user="admin.avatar"
                class="ff-venue-details__owner"
                :size="80"
              />
            </v-flex>
            <v-flex>
              <div class="primary--text subheading">
                {{ admin.name }}
              </div>
              <div class="grey--text caption">
                {{ admin.email }}
              </div>
              <div class="grey--text caption">
                {{ admin.level_name }} @{{ admin.company_name }}
              </div>
            </v-flex>
          </v-layout>
        </v-flex>
      </v-layout>
    </v-card-text>
  </v-card>
</template>
<script>
import FUserAvatar from '@freshinup/core-ui/src/components/FUserAvatar'

import MapValueKeysToData from '~/mixins/MapValueKeysToData'

export const DEFAULT_COMPANY = {
  type: 0,
  status: 0,
  name: '',
  logo: '',
  members_count: 0,
  admin: {
    name: '',
    email: '',
    company_name: '',
    avatar: '',
    level_name: ''
  }
}

export const DEFAULT_IMAGE = 'https://via.placeholder.com/800x600.png'

export default {
  components: {
    FUserAvatar
  },
  mixins: [MapValueKeysToData],
  props: {
    // overriding value prop to define default value
    value: { type: Object, default: () =>  DEFAULT_COMPANY },
    isLoading: { type: Boolean, default: false },
    types: { type: Array, default: () => [] }
  },
  data () {
    return {
      ...DEFAULT_COMPANY
    }
  },
  computed: {
    companyLogo () {
      return this.logo || DEFAULT_IMAGE
    },
    typeName () {
      return 'Supplier'
    },
    statusName () {
      return 'Active'
    },
    emptyCompany () {
      return !this.name
    }
  },
  methods: {
    viewDetails () {
      this.$emit('manage-view', this.payload)
      this.$emit('manage', 'view', this.payload)
    }
  }
}
</script>
