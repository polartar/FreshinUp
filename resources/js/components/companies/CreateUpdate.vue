<script>
import CreateUpdate from 'fresh-bus/components/pages/admin/companies/CreateUpdate.vue'

export default {
  extends: CreateUpdate,
  props: {
    isAdmin: {
      type: Boolean,
      default: false
    }
  },
  data () {
    return {
      disableStatus: true
    }
  },
  watch: {
    isAdmin () {
      this.disableStatus = !this.isAdmin
    }
  }
}
</script>
