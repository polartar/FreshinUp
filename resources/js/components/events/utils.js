import get from 'lodash/get'

const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}

export default getFileNameCopy
