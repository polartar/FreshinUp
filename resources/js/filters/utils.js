export function prefixStr (value, prefix = '@') {
  if (!value) return ''
  return value ? `${prefix} ${value}` : ''
}
