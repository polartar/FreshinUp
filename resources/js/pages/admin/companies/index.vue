<script>
import CompaniesPage from 'fresh-bus/pages/admin/companies/index.vue'
import CompaniesList from '~/components/datatable/CompaniesList.vue'

export default {
  extends: CompaniesPage,
  components: {
    CompaniesList
  }
}
</script>