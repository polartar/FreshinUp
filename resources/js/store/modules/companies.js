import makeRestStore from '@freshinup/core-ui/src/store/utils/makeRestStore'

import store from 'fresh-bus/store/modules/companies'

export default ({ items, item }) => {
  return {
    ...store,
    namespaced: true,
    modules: {
      ...store.modules,
      squareLocations: makeRestStore(
        'squareLocations',
        { items, item },
        {
          itemsPath: ({ companyId }) => `/foodfleet/companies/${companyId}/square-locations`,
          itemPath: ({ companyId, id }) => `/foodfleet/companies/${companyId}/square-locations/${id}`
        }
      )
    }
  }
}
