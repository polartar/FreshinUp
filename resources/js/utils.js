import get from 'lodash/get'
/**
 *
 * @param options HTML attributes
 * @param [options.type] string 'link'|'script' or anything in fact
 * @param [options.container] HTMLElement
 * @returns {Promise<unknown>}
 */
export const lazyLoad = (options) => {
  return new Promise((resolve, reject) => {
    if (!options.type) {
      return reject(new Error('options.type is required'))
    }
    if (!options.container) {
      options.container = document.body
    }
    const { type, container, ...attributes } = options
    let element = document.createElement(type)
    Object.keys(attributes).forEach(key => element.setAttribute(key, attributes[key]))
    element.addEventListener('load', () => {
      resolve(element)
    })
    element.addEventListener('error', (e) => {
      reject(e)
    })
    container.append(element)
  })
}

export const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}
