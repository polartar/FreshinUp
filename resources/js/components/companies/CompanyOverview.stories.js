import { storiesOf } from '@storybook/vue'
import { action } from '@storybook/addon-actions'

import CompanyOverview from './CompanyOverview.vue'

import { FIXTURE_USER_TYPES } from '../../../../tests/Javascript/__data__/userTypes'
import { FIXTURE_COMPANY } from '../../../../tests/Javascript/__data__/companies'

const methods = {
  onViewDetails (payload) {
    action('viewDetails')(payload)
  }
}

export const Empty = () => ({
  components: { CompanyOverview },
  template: `
      <v-container>
        <company-overview/>
      </v-container>
    `
})

export const Loading = () => ({
  components: { CompanyOverview },
  template: `
      <v-container>
        <company-overview
          is-loading
        />
      </v-container>
    `
})

export const Populated = () => ({
  components: { CompanyOverview },
  data () {
    return {
      company: { ...FIXTURE_COMPANY['data'] },
      companyTypes: FIXTURE_USER_TYPES
    }
  },
  methods,
  template: `
      <v-container>
        <company-overview
          :value="company"
          :types="companyTypes"
          @manage-view="onManageView"/>
      </v-container>
    `
})

storiesOf('FoodFleet|components/companies/CompanyOverview', module)
  .addParameters({
    backgrounds: [
      { name: 'default', value: '#f1f3f6', default: true }
    ]
  })
  .add('Empty', Empty)
  .add('Loading', Loading)
  .add('Populated', Populated)
