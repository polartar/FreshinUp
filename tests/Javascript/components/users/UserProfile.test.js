import { mount } from '@vue/test-utils'
import { createLocalVue } from 'tests/utils'
import { FIXTURE_USER } from 'tests/__data__/user'
import { FIXTURE_COMPANY } from 'tests/__data__/companies'
import Component from '~/components/users/UserProfile.vue'

describe('UserProfile', () => {
  let localVue
  beforeEach(() => {
    const vue = createLocalVue({ validation: true })
    localVue = vue.localVue
  })
  describe('Methods', () => {
    it('snapshot', async () => {
      const wrapper = mount(Component, {
        localVue: localVue,
        propsData: {
          user: FIXTURE_USER,
          isCurrentUser: true,
          company: FIXTURE_COMPANY
        }
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
