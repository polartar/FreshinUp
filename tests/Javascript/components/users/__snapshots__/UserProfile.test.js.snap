// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile Snapshots isCurrentUser 1`] = `
<div
  class="layout row wrap"
>
  <div
    class="flex xs12 mb-4"
  >
    <div
      class="v-card v-sheet theme--light"
    >
      <div
        class="v-card__title"
      >
        <h3
          class="grey--text"
        >
          
      Basic Information
    
        </h3>
         
        <!---->
      </div>
       
      <hr
        class="v-divider theme--light"
      />
       
      <div
        class="layout pa-3 row wrap"
      >
        <div
          class="flex xs8"
        >
          <div
            class="layout row wrap"
          >
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            First name
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Last name
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Company
          
              </div>
               
              <div
                class="pt-4"
              >
                
            
          
              </div>
            </div>
             
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Title
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <!---->
           
          <div
            class="layout row wrap"
          >
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Email
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="layout row wrap"
          >
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Office phone
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Mobile phone
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="layout row wrap"
          >
            <div
              class="flex xs3 px-1"
            >
              <button
                class="v-btn v-btn--depressed theme--light primary"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
            Change password
          
                </div>
              </button>
            </div>
          </div>
           
          <!---->
        </div>
         
        <div
          class="flex xs4 pl-4"
        >
          <div
            class="px-2 xs-12 md-4"
          >
            <div
              class="mb-2 text-uppercase grey--text font-weight-bold"
            >
              
          Profile picture
        
            </div>
             
            <input
              class="ff-profile-picture__image_input"
              type="file"
            />
             
            <div
              class="v-responsive v-image grey lighten-2"
            >
              <div
                class="v-image__image v-image__image--preload v-image__image--cover"
                mode="in-out"
                name="fade-transition"
                style="background-position: center center;"
              />
              <div
                class="v-responsive__content"
              />
            </div>
          </div>
           
          <div
            class="py-3 d-flex justify-space-between"
          >
            <button
              class="v-btn v-btn--depressed theme--light primary"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                
          Upload Image
        
              </div>
            </button>
             
            <button
              class="v-btn v-btn--disabled v-btn--depressed theme--light"
              disabled="disabled"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                
          Delete Image
        
              </div>
            </button>
          </div>
        </div>
      </div>
       
      <hr
        class="v-divider theme--light"
      />
       
      <div
        class="layout pa-3"
      >
        <div
          class="flex"
        >
          <button
            class="v-btn v-btn--depressed theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
        Cancel
      
            </div>
          </button>
           
          <button
            class="v-btn v-btn--depressed theme--light primary"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
        Save changes
      
            </div>
          </button>
        </div>
         
        <div
          class="flex text-xs-right"
        >
          <button
            class="v-btn v-btn--depressed theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
        Delete account
      
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="flex xs12 mb-4"
  >
    <div
      class="v-card v-sheet theme--light"
    >
      <div
        class="v-card__title"
      >
        <h3
          class="grey--text"
        >
          
      Notification settings
    
        </h3>
         
        <!---->
      </div>
       
      <hr
        class="v-divider theme--light"
      />
       
      <div
        class="layout pa-4 row wrap"
      >
        <div>
          Coming Soon
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserProfile Snapshots isCurrentUserAdmin 1`] = `
<div
  class="layout row wrap"
>
  <div
    class="flex xs12 mb-4"
  >
    <div
      class="v-card v-sheet theme--light"
    >
      <div
        class="v-card__title"
      >
        <h3
          class="grey--text"
        >
          
      Basic Information
    
        </h3>
         
        <!---->
      </div>
       
      <hr
        class="v-divider theme--light"
      />
       
      <div
        class="layout pa-3 row wrap"
      >
        <div
          class="flex xs8"
        >
          <div
            class="layout row wrap"
          >
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            First name
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Last name
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Company
          
              </div>
               
              <div
                class="pt-4"
              >
                
            
          
              </div>
            </div>
             
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Title
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <!---->
           
          <div
            class="layout row wrap"
          >
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Email
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="layout row wrap"
          >
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Office phone
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Mobile phone
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="layout row wrap"
          >
            <div
              class="flex xs3 px-1"
            >
              <button
                class="v-btn v-btn--depressed theme--light primary"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
            Change password
          
                </div>
              </button>
            </div>
          </div>
           
          <!---->
        </div>
         
        <div
          class="flex xs4 pl-4"
        >
          <div
            class="px-2 xs-12 md-4"
          >
            <div
              class="mb-2 text-uppercase grey--text font-weight-bold"
            >
              
          Profile picture
        
            </div>
             
            <input
              class="ff-profile-picture__image_input"
              type="file"
            />
             
            <div
              class="v-responsive v-image grey lighten-2"
            >
              <div
                class="v-image__image v-image__image--preload v-image__image--cover"
                mode="in-out"
                name="fade-transition"
                style="background-position: center center;"
              />
              <div
                class="v-responsive__content"
              />
            </div>
          </div>
           
          <div
            class="py-3 d-flex justify-space-between"
          >
            <button
              class="v-btn v-btn--depressed theme--light primary"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                
          Upload Image
        
              </div>
            </button>
             
            <button
              class="v-btn v-btn--disabled v-btn--depressed theme--light"
              disabled="disabled"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                
          Delete Image
        
              </div>
            </button>
          </div>
        </div>
      </div>
       
      <hr
        class="v-divider theme--light"
      />
       
      <div
        class="layout pa-3"
      >
        <div
          class="flex"
        >
          <button
            class="v-btn v-btn--depressed theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
        Cancel
      
            </div>
          </button>
           
          <button
            class="v-btn v-btn--depressed theme--light primary"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
        Save changes
      
            </div>
          </button>
        </div>
         
        <div
          class="flex text-xs-right"
        >
          <button
            class="v-btn v-btn--depressed theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
        Delete account
      
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="flex xs12 mb-4"
  >
    <div
      class="v-card v-sheet theme--light"
    >
      <div
        class="v-card__title"
      >
        <h3
          class="grey--text"
        >
          
      Notification settings
    
        </h3>
         
        <!---->
      </div>
       
      <hr
        class="v-divider theme--light"
      />
       
      <div
        class="layout pa-4 row wrap"
      >
        <div>
          Coming Soon
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserProfile Snapshots without user and company 1`] = `
<div
  class="layout row wrap"
>
  <div
    class="flex xs12 mb-4"
  >
    <div
      class="v-card v-sheet theme--light"
    >
      <div
        class="v-card__title"
      >
        <h3
          class="grey--text"
        >
          
      Basic Information
    
        </h3>
         
        <!---->
      </div>
       
      <hr
        class="v-divider theme--light"
      />
       
      <div
        class="layout pa-3 row wrap"
      >
        <div
          class="flex xs8"
        >
          <div
            class="layout row wrap"
          >
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            First name
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Last name
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Company
          
              </div>
               
              <div
                class="pt-4"
              >
                
            
          
              </div>
            </div>
             
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Title
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <!---->
           
          <div
            class="layout row wrap"
          >
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Email
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="layout row wrap"
          >
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Office phone
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="flex xs6 px-2"
            >
              <div
                class="mb-2 text-uppercase grey--text font-weight-bold"
              >
                
            Mobile phone
          
              </div>
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-text-field__slot"
                    >
                      <input
                        placeholder="Name"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="layout row wrap"
          >
            <div
              class="flex xs3 px-1"
            >
              <button
                class="v-btn v-btn--depressed theme--light primary"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
            Change password
          
                </div>
              </button>
            </div>
          </div>
           
          <!---->
        </div>
         
        <div
          class="flex xs4 pl-4"
        >
          <div
            class="px-2 xs-12 md-4"
          >
            <div
              class="mb-2 text-uppercase grey--text font-weight-bold"
            >
              
          Profile picture
        
            </div>
             
            <input
              class="ff-profile-picture__image_input"
              type="file"
            />
             
            <div
              class="v-responsive v-image grey lighten-2"
            >
              <div
                class="v-image__image v-image__image--preload v-image__image--cover"
                mode="in-out"
                name="fade-transition"
                style="background-position: center center;"
              />
              <div
                class="v-responsive__content"
              />
            </div>
          </div>
           
          <div
            class="py-3 d-flex justify-space-between"
          >
            <button
              class="v-btn v-btn--depressed theme--light primary"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                
          Upload Image
        
              </div>
            </button>
             
            <button
              class="v-btn v-btn--disabled v-btn--depressed theme--light"
              disabled="disabled"
              type="button"
            >
              <div
                class="v-btn__content"
              >
                
          Delete Image
        
              </div>
            </button>
          </div>
        </div>
      </div>
       
      <hr
        class="v-divider theme--light"
      />
       
      <div
        class="layout pa-3"
      >
        <div
          class="flex"
        >
          <button
            class="v-btn v-btn--depressed theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
        Cancel
      
            </div>
          </button>
           
          <button
            class="v-btn v-btn--depressed theme--light primary"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
        Save changes
      
            </div>
          </button>
        </div>
         
        <div
          class="flex text-xs-right"
        >
          <button
            class="v-btn v-btn--depressed theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
        Delete account
      
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="flex xs12 mb-4"
  >
    <div
      class="v-card v-sheet theme--light"
    >
      <div
        class="v-card__title"
      >
        <h3
          class="grey--text"
        >
          
      Notification settings
    
        </h3>
         
        <!---->
      </div>
       
      <hr
        class="v-divider theme--light"
      />
       
      <div
        class="layout pa-4 row wrap"
      >
        <div>
          Coming Soon
        </div>
      </div>
    </div>
  </div>
</div>
`;
